File: Reverse_Words_In_A_String/py.d/reverse_words_string.py
class Solution:
  def reverseWords(self, s: str) -> str:
    # return " ".join(reversed(s.split())) # more concise; but slower in python3
    s = s.split()
    s.reverse()
    return " ".join(s)

if __name__=="__main__":
  sol = Solution()
  print(sol.reverseWords("the sky is blue")) # => "blue is sky the"

File: Reverse_Words_In_A_String/js.d/reverse_words_string.js

var reverseWords = function(s){
  s = s.trimEnd();
  s = s.trimStart();
  s = s.replace(/\s{2,}/g, ' ');
  let str_array = s.split(" ");
  str_array = str_array.reverse();
  return str_array.join(" ");
};

var test_strings = [
  "  a good  example  ",
  "the sky is blue",
  "  hello world!  ",
  " lots  of     spaces here    ",
  "1 ",
  " ",
  "",
  "  ",
  "    ",
  "a"
];

test_strings.map(x => console.log("_"+reverseWords(x)+"_"));

/* - Optimal solution
var reverseWords = function(s) {
   s=s.trim().replace(/(?=\s)(\s)(?=\s)/g,'') 
   return s.split(' ').reverse().join(' ')
};
*/
